### ISSUE TITLE
Investigate possible performance issues

### ISSUE THREAD

**Issue by let4be** [ZeroK-RTS/Zero-K#2148]
I have custom configured springsettings.cfg and in-game widgets(performance offenders disabled after profiling)

My game performance is above average, everything runs smoothly even in large games(8v8 and above)
What is more interesting than I was able to achieve this playing on very old GPU(2007y old), though my CPU is decent enough(i7 2600k) and I have 32gb of DDR3 RAM

So in big games I have pretty stable 30+ FPS while a lot of people experience very bad performance with FPS floating at 1-5... So I wonder why is this and what can we do about this

Here are my configs
https://pastebin.com/eN5KNSrn
https://pastebin.com/jea2iWMA
https://pastebin.com/S0iMD5t2

**Comment by GoogleFrog** [comment]
Widget activation diff:
 * AllyCursors -> Off
 * Auto-toggle false color vision -> Off
 * CameraShake -> Off
 * Deferred rendering -> Off
 * Depth of Field Shader -> Off
 * Ghost Site -> Off
 * Grab Input -> Off
 * HUD Presets -> Off
 * Lups -> Off
 * Lups Manager -> Off
 * Map Edge Extension -> Off
 * MessageBoxes ZK -> Off
 * Mission Results Handler -> Off
 * Projectile Lights -> Off
 * Set Springsettings -> Off
 * SmoothCam -> On
 * Sun and Atmosphere Handler. -> Off
 * Units on Fire -> Off
 * Voice Assistant -> Off
 * Win Counter -> Off

**Comment by let4be** [comment]
Ally cursors and Map Edge Extension were disabled manually right after I did profiling, so I'd suggest to start checking from them

**Comment by sprunk** [comment]
>I was able to achieve this playing on very old GPU, though my CPU is decent enough

As far as I know Spring's main bottleneck is CPU by far.

**Comment by let4be** [comment]
I agree. I'm still curious why I do not lag while a lot of people crying about 1 FPS...

**Comment by GoogleFrog** [comment]
@let4be comment on a large game (8v8ish), or post it here, with periods of action and 800 to 1000 units on the map at points. I'll make a savestate off it and see if I can incorporate a benchmark.

**Comment by sprunk** [comment]
Nothing here seems to be valuable anymore. Attached configs are almost 7 years old and there are fresher perf related tickets.
